comp/process/agent/agentimpl/agent.go

```diff
@@ -70,37 +70,21 @@ func newProcessAgent(p processAgentParams) optional.Option[agent.Component] {
 	// Look to see if any checks are enabled, if not, return since the agent doesn't need to be enabled.
 	if len(enabledChecks) == 0 {
 		p.Log.Info(agentDisabledMessage)
-		return optional.NewNoneOption[agent.Component]()
+		return processAgent{
+			enabled: false,
+		}
 	}
 
 	processAgentComponent := processAgent{
-		Checks: enabledChecks,
-		Log:    p.Log,
-		Runner: p.Runner,
-	}
-
-	p.Lc.Append(fx.Hook{
```

Comment:
Lifecycles still being removed? Should instead they be conditionally added based upon the `enabled` state?

Reply:
So this component no longer has anything to run. It has become a wrapper for the all the process-agent dependencies. The `runner` is our "scheduler" responsible for running checks at specified intervals. The `submitter` is responsible for sending the check results to the backend.